### PR TITLE
[main] include exception type

### PR DESCRIPTION
**Pull request by YuanTingHsieh**
### Description

- Include exception type
- Fix unit test for https://github.com/NVIDIA/NVFlare/pull/1742 by @taleinat 

By Tal:

The `str()` of an exception normally doesn't include its type. Including the type of the exception is very useful.

In the rare cases of exceptions with no details in their string representation, this is especially significant.

This is similar to the format used in Python exception tracebacks:

```python
>>> int("a")
Traceback (most recent call last):
  File "<pyshell#>", line 1, in <module>
ValueError: invalid literal for int() with base 10: 'a'

>>> try:
	int("a")
except ValueError as exc:
	print(str(exc))

invalid literal for int() with base 10: 'a'
```

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: nvflare/security/logging.py [hunk 1]
```diff
@@ -130,4 +130,4 @@ def secure_format_exception(e: Exception) -> str:
     if is_secure():
         return str(type(e))
     else:
-        return str(e)
+        return f"{str(type(e))}: {str(e)}"
```
Comment (taleinat) [inline nvflare/security/logging.py:133]: I've recently noticed this doesn't print what I expected, and should be fixed thus:
```suggestion
        return f"{type(e).__name__}: {str(e)}"
```

File: tests/unit_test/fuel/utils/component_builder_test.py [hunk 2]
```diff
@@ -77,9 +77,9 @@ def test_component_failure(self):
 
         # the failure message changes since 3.10
         if is_python_greater_than_309():
-            msg = "Class nvflare.app_common.np.np_model_locator.NPModelLocator has parameters error: NPModelLocator.__init__() got an unexpected keyword argument 'xyz'."
+            msg = "Class nvflare.app_common.np.np_model_locator.NPModelLocator has parameters error: <class 'TypeError'>: NPModelLocator.__init__() got an unexpected keyword argument 'xyz'."
```
Comment (taleinat) [inline tests/unit_test/fuel/utils/component_builder_test.py:80]:
```suggestion
            msg = "Class nvflare.app_common.np.np_model_locator.NPModelLocator has parameters error: TypeError: NPModelLocator.__init__() got an unexpected keyword argument 'xyz'."
```

Comment (taleinat) [reply tests/unit_test/fuel/utils/component_builder_test.py:80]: Similarly in the other tests.

File: nvflare/security/logging.py (hunk 1 above)
Comment (YuanTingHsieh) [reply nvflare/security/logging.py:133]: thanks, I will change accordingly

**Review by taleinat — APPROVED** [review]
LGTM

**Pull request merged**